src/main/java/org/rumbledb/runtime/functions/FunctionItemCallIterator.java

```diff
@@ -185,7 +185,7 @@ private FunctionRuntimeIterator generatePartiallyAppliedFunction(DynamicContext
                 RDDArgumentValues,
                 DFArgumentValues
         );
-        return new FunctionRuntimeIterator(partiallyAppliedFunction, ExecutionMode.LOCAL, getMetadata());
+        return new ConstantRuntimeIterator(partiallyAppliedFunction, ExecutionMode.LOCAL, getMetadata());
```

Comment:
Third fix: after computing a partially applied function, the iterator we return is only a wrapper on the generated function item, which must be returned *as is*. This is not an inline function expression evaluation.